packages/blob/src/client.ts

```diff
@@ -46,12 +46,34 @@ export const put = createPutMethod<ClientPutCommandOptions>({
 // upload()
 // This is a client-side wrapper that will fetch the client token for you and then upload the file
 export interface UploadOptions {
+  /**
+   * Whether the blob should be publicly accessible. Support for private blobs is planned.
+   */
   access: 'public';
+  /**
+   * Defines the media type of the blob. By default, this value is inferred from the pathname.
+   */
   contentType?: string;
+  /**
+   * A route that implements the `handleUpload` function for generating a client token.
+   */
   handleUploadUrl: string;
+  /**
+   * Additional data which will be sent to your `handleUpload` route.
+   */
   clientPayload?: string;
 }
 
+/**
+ * Uploads a blob into your store from the client.
+ * Detailed documentation can be found here: https://vercel.com/docs/storage/vercel-blob/using-blob-sdk#client-uploads
+ *
+ * If you want to upload from your server instead, check out the documentation for the put operation: https://vercel.com/docs/storage/vercel-blob/using-blob-sdk#upload-a-blob
+ *
+ * @param pathname - The pathname to upload the blob to. This includes the filename.
+ * @param body - The contents of your blob. This has to be a supported fetch body type https://developer.mozilla.org/en-US/docs/Web/API/fetch#body.
+ * @param options - Additional options.
```

Comment:
Do we want to document said options here?

Reply:
## This is visible on hovering `.put`:
<img width="442" alt="Screenshot 2023-10-23 at 16 52 14" src="https://github.com/vercel/storage/assets/41523937/211b7c7e-5baa-4074-8b7c-3f0b3d8491c8">

--- 

## This is shown in the autocomplete dropdown:
<img width="817" alt="Screenshot 2023-10-23 at 16 52 27" src="https://github.com/vercel/storage/assets/41523937/485867be-6260-452f-9ba7-07a21514a76c">

--- 

## And this is shown without forcing the autocomplete dropdown by just start typing:
<img width="821" alt="Screenshot 2023-10-23 at 16 52 35" src="https://github.com/vercel/storage/assets/41523937/d8e016f1-425e-4ae1-bbc1-d77d26622255">